x-pack/apm-server/main.go

```diff
@@ -230,14 +246,33 @@ func closeBadger() error {
 	return nil
 }
 
+func closeStorage() (err error) {
+	if storage != nil {
+		// Force flush without limit consideration
+		err = storage.Flush(0)
```

Comment:
Do we need to flush here? I would expect it to be the responsibility of the processor to flush when it exits.

Reply:
Good point, while writing the code I decided that flush should be handled globally but with the introduction of wrapped read writers residing in the Processor lifecycle, it does make more sense to flush the processor on `Stop`. Let me make the required changes.